src/ray/raylet/scheduling/local_resource_manager.h

```diff
@@ -193,8 +206,10 @@ class LocalResourceManager : public syncer::ReporterInterface {
   scheduling::NodeID local_node_id_;
   /// Resources of local node.
   NodeResourceInstances local_resources_;
+
   /// A map storing when the resource was last idle.
-  absl::flat_hash_map<scheduling::ResourceID, absl::optional<absl::Time>>
+  absl::flat_hash_map<std::variant<WorkFootprint, scheduling::ResourceID>,
```

Comment:
spoke with victoria offline: 
i think it makes sense to combine all these into single map, if the map's sole purpose is to determine a node is idle or not. however i'd suggest we unify the `std::variant` and `WorkFootprint` into a single data structure.

Reply:
Aliased, renamed, and added descriptive comments